lib/galaxy/datatypes/binary.py

```diff
@@ -835,12 +835,12 @@ def set_meta(self, dataset, overwrite=True, **kwd):
         super(Loom, self).set_meta(dataset, overwrite=overwrite, **kwd)
         try:
             with h5py.File(dataset.file_name) as loom_file:
```

Comment:
```suggestion
            with h5py.File(dataset.file_name, 'r') as loom_file:
```

otherwise the file is locked.

The same should be fixed at lines 814, 883 and 1006.